Sources/Git/Commit.swift

```diff
@@ -0,0 +1,136 @@
+
+    /// The committer of the commit (user who commited).
+    public lazy var committer: Signature? = {
+        return git_commit_committer(handle).map(Signature.init(pointer:))
+    }()
+
+    /// The author of the commit (user who wrote the code).
+    public lazy var author: Signature? = {
+        return git_commit_author(handle).map(Signature.init(pointer:))
+    }()
+
+    /// The number of parents of the commit.
+    public lazy var parentCount: Int = {
+        return Int(git_commit_parentcount(handle))
+    }()
+
+    /// The full message of the commit.
+    public lazy var message: String = {
+        return String(cString: git_commit_message(handle))
+    }()
+
+    init(handle: Handle, objectID: ObjectID? = nil) {
+        self.handle = handle
+
+        if let objectID = objectID {
+            self.objectID = objectID
+        }
+    }
+
+    // Get parent commit, or nil if this is the first commit.
+    public func getParent() throws -> Commit? {
+        guard parentCount > 0 else { return nil }
+
+        var parentHandle: Handle?
+        try wrap { git_commit_parent(&parentHandle, self.handle, 0) }
+        return Commit(handle: parentHandle!)
+    }
+
+    // Get commit tree.
+    public func getTree() throws -> Tree {
+        var treeHandle: Tree.Handle?
+        try wrap { git_commit_tree(&treeHandle, handle) }
+        return Tree(handle: treeHandle!)
+    }
+
+    deinit {
+        git_commit_free(handle)
+    }
+}
+
+extension Commit: Hashable {
+    public static func == (lhs: Commit, rhs: Commit) -> Bool {
+        return lhs.objectID == rhs.objectID
+    }
+
+    public var hashValue: Int {
+        return objectID.hashValue
+    }
+}
+
+/// An action signature (e.g. for committers, taggers, etc)
+public struct Signature {
+    /// Pointer to the git signature struct (doesn't need to be freed).
+    private let pointer: UnsafePointer<git_signature>
+
+    /// User name
+    public lazy var name: String = {
+        return String(cString: pointer.pointee.name)
+    }()
+
+    /// User email
+    public lazy var email: String = {
+        return String(cString: pointer.pointee.email)
+    }()
+
+    init(pointer: UnsafePointer<git_signature>) {
+        self.pointer = pointer
+    }
+}
+
+public final class RevisionIterator: IteratorProtocol, Sequence {
+
+    private let walker: OpaquePointer
+
+    init(from oid: git_oid, handle: Repository.Handle) throws {
+        var walker: OpaquePointer?
+        try wrap { git_revwalk_new(&walker, handle) }
+        git_revwalk_sorting(walker, GIT_SORT_TOPOLOGICAL.rawValue)
+        var oid = oid
+        git_revwalk_push(walker, &oid)
+        self.walker = walker!
+    }
+
+    deinit {
+        git_revwalk_free(walker)
+    }
+
+    public func next() -> ObjectID? {
+        var oid = git_oid()
+        let ret = git_revwalk_next(&oid, walker)
+
+        // Check if iterator has reached the end.
+        if ret == GIT_ITEROVER.rawValue {
+            return nil
+        }
```

Comment:
I bet we could make use of a helper function which would elevate the iterator functions return value to an optional. That would simplify this, and some places below where there is a while true loop which could become `while let value = ...`

You could also probably find a way to just make them all into a real sequence.